### PR TITLE
Adds CommandCode enum and replaces use of TPM2_CC.

### LINKED ISSUE #296: Create native type for the TPM2_CC constants.
A type for the constants in TPM2_CC should be defined in order to remove the usage of TPM2_CC from the context methods API. This could be done in a similar way as the algorithm constants was created.

### PR DESCRIPTION
**Pull request by Superhepper**
This fixes #296

- Adds the CommandCode enum.
- Replace the use of TPM2_CC with the CommandCode enum.

### CODE REVIEW

**Comment by wiktor-k** [comment]
This is very nice and cleans up the duplicate code too (thanks!). :+1: 

One thing I'm worried about: if there are vendor-specific commands don't we need an `Unknown(u32)`/`VendorSpecific(u32)` variant in the enum? Or at least marking the enum with `#[non_exhaustive]` to let others know that we can add more later?

**Comment by Superhepper** [comment]
Vendor specific commands will result in a Unsupported error right now.

**Review by ionut-arm — APPROVED** [review]
Thanks for the patch!

We can probably raise an issue (if we don't already have one) to track the vendor-specific stuff.

File: tss-esapi/src/constants/tss.rs [hunk 1]
```diff
@@ -69,7 +69,7 @@ pub const TPM2_CC_ClockSet: TPM2_CC = 0x00000128;
 pub const TPM2_CC_HierarchyChangeAuth: TPM2_CC = 0x00000129;
 pub const TPM2_CC_NV_DefineSpace: TPM2_CC = 0x0000012a;
 pub const TPM2_CC_PCR_Allocate: TPM2_CC = 0x0000012b;
-pub const TPM2_CC_PCR_SetAuthPolicy: TPM2_CC = 0x0F000012c;
+pub const TPM2_CC_PCR_SetAuthPolicy: TPM2_CC = 0x00000012c;
```
[the remark this answers is not in the document]
Comment (Superhepper) [reply tss-esapi/src/constants/tss.rs:72]: No idea why it was wrong in the first place. But my tests caught it!

**Review by wiktor-k — APPROVED** [review]
LGTM 👍 Thanks!

**Pull request merged**